### PR TITLE
WIP - Add missing rules to RULES.md

### LINKED ISSUE #758: Add missing rules to RULES.md
As discussed in #741 

Add missing rules from `eslintrc.json` to RULES.md. I am about halfway done with this. Opening issue for discussion, feedback, etc.

### PR DESCRIPTION
**Pull request by joshuacolvin**
Currently have 55 of the 102 missing rules added. Would love some feedback/guidance on descriptions, examples, and order.

Fixes #758

### CODE REVIEW

**Comment by dcousens** [comment]
Couldn't this just be derived and purely be links to the eslint documentation? (minus the inline examples, of course)

**Comment by joshuacolvin** [comment]
Started working on this based on the conversation with @feross here: #741

I could go either way. One concern I have is that adding descriptions and examples for each rule will make RULES.md excessively long. 

I will hold off until we have a consensus on the right way to proceed.

**Comment by feross** [comment]
@joshuacolvin This PR is excellent! I think this is the right way to proceed, so please continue adding the rest of the rules if you're able to.

The file will get pretty long, this is true. But we can work on putting the rules into different categories to make it easier to find the right section.

Eventually, I would like to get RULES.md to a point where it's similar to https://github.com/airbnb/javascript#table-of-contents Basically a list of all the rules, as well as our rationale for each one.

**Comment by joshuacolvin** [comment]
@feross Thanks! I will proceed.

**Comment by feross** [comment]
@joshuacolvin Please ping me when this PR is ready for final review

**Comment by joshuacolvin** [comment]
@feross will do. Should have this finished up this weekend.

**Comment by feross** [comment]
@joshuacolvin Awesome, thanks!

**Comment by joshuacolvin** [comment]
@feross ready for final review!

File: RULES.md [hunk 1]
```diff
@@ -251,6 +251,1015 @@ your code.
...
+  const user = {
+    name: 'Jane Doe', age: 30,
+    username: 'jdoe86'            // ✗ avoid
+  }
+
+  const user = { name: 'Jane Doe', age: 30, username: 'jdoe86' }    // ✓ ok
+  const user = {
+    name: 'Jane Doe',
+    age: 30,
+    username: 'jdoe86
```
Comment (feross) [inline RULES.md]: You're missing a quote here

Comment (joshuacolvin) [inline RULES.md]: Sorry about that! Fixed.

**Comment by feross** [comment]
Thanks for the PR @joshuacolvin, I just merge it and added you as a collaborator. Please read the [contributor guidelines](https://github.com/feross/standard/blob/master/CONTRIBUTING.md).

**Comment by joshuacolvin** [comment]
@feross No problem. Thanks! I will read the guidelines.

**Pull request merged**